File: typescript/base-hmi-types/src/screens/controls/HmiMediaControl.ts
import { HmiProperty } from "../base.js";
import { HmiControlWindowBase } from "../base/HmiControlWindowBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";

export class HmiMediaControl extends HmiControlWindowBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiMediaControl;
  }

  source?: HmiProperty<string>;
  autoPlay?: HmiProperty<boolean>;
}

File: src/BaseHmiTypes/Screens/Controls/HmiMediaControl.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public class HmiMediaControl : HmiControlWindowBase
{
    public HmiMediaControl()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiMediaControl;
    }

    public HmiProperty<string>? Source { get; set; }

    public HmiProperty<bool>? AutoPlay { get; set; }
}
